### DISCUSSION TITLE
-bordercolor not working

### DISCUSSION THREAD

**Discussion by ak47twq** [ImageMagick/ImageMagick#4449] [Help] [answered]
ImageMagick 7 on win 10
```
magick -size 8000x2000 xc:red2 test.jpg 
magick test.jpg -pointsize 720 -fill yellow2 -annotate +100+1800 "NFL WEEK 9 PICKS" -trim +repage  -border 500 -bordercolor red2 testfont.jpg
```
![testfont](https://user-images.githubusercontent.com/72854702/140611517-58986355-b00b-4f77-aa84-b3073a6624db.jpg)

why the border has no color?
what part of the code should i change？
thx!!!

**Comment by GeeMack** [comment, marked as answer]
ImageMagick v7 is very picky about the syntax. In your command you'll need to set the border color with `-bordercolor red2` *before* you add the border to the image with `-border 500`.

**Reply by ak47twq** [reply]
thx
```
magick -size 8000x2000 xc:red2 test.jpg && magick test.jpg -pointsize 720 -fill yellow2 -annotate +100+1800 "NFL WEEK 9 PICKS" -trim +repage -bordercolor red2 -border 500  testfont.jpg
```
![testfont](https://user-images.githubusercontent.com/72854702/140628232-4d9826e3-eaa9-48d2-92a9-4de7d6a3b8ed.jpg)

it works!

**Comment by snibgo** [comment]
@ak47twq: IM commands should be given in the order you want them executed. "-border" is an operation that will draw a border. "-bordercolor" is a setting that defines what colour any _future_ "-border" commands should use.

If you define the colour _after_ the border is draw, that is too late.

**Reply by ak47twq** [reply]
thank you!
